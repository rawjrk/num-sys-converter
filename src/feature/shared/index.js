import Form from "./components/Form";
import Label from "./components/Label";
import Input from "./components/Input";
import Select from "./components/Select";
import Options, { optionsPropType } from "./components/Options";
import Button from "./components/Button";
import Footer from "./components/Footer";
import Link from "./components/Link";

export {
  Form,
  Label,
  Input,
  Select,
  Options,
  optionsPropType,
  Button,
  Footer,
  Link,
};
