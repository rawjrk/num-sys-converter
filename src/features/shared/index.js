import Button from "./components/Button";
import Footer from "./components/Footer";
import Link from "./components/Link";

export { Button, Footer, Link };
export { Form, Label, Input, Select, Options } from "./components/form";
