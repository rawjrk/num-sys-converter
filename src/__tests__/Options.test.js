import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Options } from "../features/shared";

const options = [
  { title: "Opt. 1", value: 1 },
  { title: "Opt. 2", value: 2 },
  { title: "Opt. 3", value: 3 },
];

describe("component Options", () => {
  it("should render successfully", () => {
    render(<Options options={options} />);

    expect(screen.getByText(/Opt. 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Opt. 2/i)).toBeInTheDocument();
    expect(screen.getByText(/Opt. 3/i)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Options options={options} />);
    expect(tree).toMatchSnapshot();
  });
});
