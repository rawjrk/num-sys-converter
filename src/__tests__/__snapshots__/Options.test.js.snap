// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Options matches snapshot 1`] = `
[
  <option
    id={1}
    value={1}
  >
    Opt. 1
  </option>,
  <option
    id={2}
    value={2}
  >
    Opt. 2
  </option>,
  <option
    id={3}
    value={3}
  >
    Opt. 3
  </option>,
]
`;
